### ISSUE TITLE
[Bug] Add regex to 'Link to Work' field

### ISSUE THREAD

**Issue by soustab10** [creativecommons/chooser#452]
## Description
Add regex check in 'Add to Work' field as it prevents users from entering random/incorrect urls.

## Reproduction

1.  Step 1: Go to `https://chooser-beta.creativecommons.org/` 
2.  Step 2 : Fill the 'License Chooser' till Attribution Details'
3. Step 3 : Fill anything random in the URLs panel.
4. See that instead of giving a prompt/warning it accepts the data.

## Expectation
The user should be prompted to enter a valid URL.

## Screenshots

![image](https://user-images.githubusercontent.com/95632583/222460246-56c3f580-80df-414c-b9a9-4d04abc403e8.png)
![image](https://user-images.githubusercontent.com/95632583/222460296-cda4688a-1615-4275-ba0d-27938116fad2.png)

## Environment

NA

## Additional context

NA

## Resolution

- [x] I would be interested in resolving this bug.

**Comment by possumbilities** [comment]
@soustab10 this is an incredibly good find and look forward to your PR!

**Comment by HoneyTyagii** [comment]
@soustab10  Hey! I noticed that the issue #452 is still open. I'd love to help out and contribute to the project.

**Comment by soustab10** [comment]
@HoneyTyagii I am working on it. Thank you.

**Comment by Ankitsharma991** [comment]
@soustab10 can you add me (Sharmaa7861@gmail.com) to the slack channel, I am not able to join don't know why. I am interested in to contribute at CC through GSoC.

**Comment by soustab10** [comment]
> @soustab10 this is an incredibly good find and look forward to your PR!

@possumbilities I was thinking of adding a check to the "Year of Creation" field as well for users to enter only a valid year ( and not something like 12345). So, can you tell me what year range should I put in? Thank you.

**Comment by soustab10** [comment]
ok then I will implement that. Thanks for helping me @possumbilities

**Comment by Cronus1007** [comment]
@soustab10 Thanks for raising this issue.
But I had a very great concern in this issue.
@possumbilities Please have a look upon this since this issue has been discussed several times. 

> Form field validation is necessary before the data is sent to the server to prevent attacks such as SQL injection. Chooser 
> does not send any information to the server, so it is not necessary. These values are only used to create the license code on 
> the 'Mark Your Work' section.
> We have already discussed sanitizing user input in [177](https://github.com/creativecommons/chooser/issues/177), so you can find 
> more context there.

@possumbilities @soustab10 to elaborate, a single user is consuming their own input + output in one sitting, only on the client-side, so we don't need to validate the information at all. What they type is what they'll get. Sorry for pinging this thing so late.

**Comment by soustab10** [comment]
@Cronus1007 In the implementation of this issue, the user can still proceed with the process even if he enters an invalid URL as the `Done` button is not disabled. I think it is better to warn the user about the invalid URL and if required , change the warning message  as well.

**Comment by possumbilities** [comment]
@soustab10 I would agree that if we are to do some level of UX validation shaping that it should perhaps impact the flow and warning message that the end-user interfaces with to align with that.

Other thoughts are that we could do some helpful nudging/shaping in a more subtle way. 

Consider this scenario: 

A user pastes in a URL minus the https/http at the beginning. Should the form check and prepend that at the beginning for them, or give them a warning it's missing?

I would lean towards checking if it's a link, and if it's missing the `http` or `https`, then the chooser prepends it, and avoids throwing an error at all. In this way there is a validation flow happening subtly under the hood that's shaping the output without prompting extra effort from the user. Yes, the placeholder text includes the https, but that doesn't necessarily mean users will follow that pattern.

**Comment by Cronus1007** [comment]
@possumbilities Sure I got your point.

**Comment by soustab10** [comment]
> @soustab10 I would agree that if we are to do some level of UX validation shaping that it should perhaps impact the flow and warning message that the end-user interfaces with to align with that.
> 
> Other thoughts are that we could do some helpful nudging/shaping in a more subtle way.
> 
> Consider this scenario:
> 
> A user pastes in a URL minus the https/http at the beginning. Should the form check and prepend that at the beginning for them, or give them a warning it's missing?
> 
> I would lean towards checking if it's a link, and if it's missing the `http` or `https`, then the chooser prepends it, and avoids throwing an error at all. In this way there is a validation flow happening subtly under the hood that's shaping the output without prompting extra effort from the user. Yes, the placeholder text includes the https, but that doesn't necessarily mean users will follow that pattern.

I tested my PR code with the above scenario and in that case neither the user gets an error, nor is there a problem accessing the link from the final text. It seems browsers do it automatically.

Here's a screenshot:
![image](https://user-images.githubusercontent.com/95632583/223783762-c6e7723b-e12b-4dbf-8ace-1df323ae97ef.png)

**Comment by possumbilities** [comment]
@soustab10 I wouldn't rely on a browser to just correct a behavior for a non-standard URL or link, but it does seem there is some shaping happening at the moment to the generated html to help account for a leading `http` or not.

For example. If I type or paste in `creativecommons.org` into the field, the html generated is as follows: 

`<a property="dct:title" rel="cc:attributionURL" href="http://creativecommons.org">` 

It's adding the `http://` on since I left it off, which is good.


If instead I type in any url that begins with `http`. For example: `httprotocol.org` I get: 

`<a property="dct:title" rel="cc:attributionURL" href="httprotocol.org">`

That's not a valid URL. This is likely an outcome of the validation checking for `http` as a string alone, prepending the input. 

It's certainly an edge case, but I think it illustrates the variability on shaping things subtley and to what degree, and what the trade-offs may be as a result.

**Comment by yumi520** [comment]
@possumbilities Hi! Seems like this issue is resolved. I tested it on my browser (Firefox). There wasn't a PR linked, so can this ticket be closed? Thanks!

**Comment by possumbilities** [comment]
@yumi520 you are correct, thank you. I've linked the relevant PR. Closing.

**Comment by possumbilities** [comment]
@TechGenie-awake if you respond here I can assign you to this as well.

**Comment by TechGenie-awake** [comment]
@possumbilities Yes, I’ll be happy to work on it!

**Comment by possumbilities** [comment]
@TechGenie-awake I *think* that the linked PR should have resolved this, but feel free to give it a second look to make sure:
* https://github.com/creativecommons/chooser/pull/611
